api/datadoghq/common/datadogpodautoscaler_types.go

```diff
@@ -167,26 +173,83 @@ type DatadogPodAutoscalerContainerResourceObjective struct {
 	Container string `json:"container"`
 }
 
+// DatadogPodAutoscalerControllerObjective defines a controller-level objective
+// +kubebuilder:object:generate=true
+type DatadogPodAutoscalerControllerObjective struct {
+	Query DatadogPodAutoscalerTimeseriesFormulaRequest `json:"query"`
+	// Value is the value of the objective
+	Value DatadogPodAutoscalerObjectiveValue `json:"value"`
+}
+
+// DatadogPodAutoscalerTimeseriesFormulaRequest is a subset of the v2 timeseries query API (metrics only).
+// It mirrors the OpenAPI "DatadogPodAutoscalerTimeseriesFormulaRequestAttributes" fields relevant to autoscaling.
+// Reference: https://github.com/DataDog/datadog-api-spec/blob/94d1542b31ad0df1da915bae84686b13ba1a65ae/spec/v2/query.yaml#L124
+// +kubebuilder:object:generate=true
+type DatadogPodAutoscalerTimeseriesFormulaRequest struct {
+	// Formulas to compute (optional).
+	// +optional
+	Formulas []DatadogPodAutoscalerQueryFormula `json:"formulas,omitempty"`
+	// +kubebuilder:validation:MinItems=1
+	Queries []DatadogPodAutoscalerTimeseriesQuery `json:"queries"`
+}
+
+// +kubebuilder:object:generate=true
+type DatadogPodAutoscalerQueryFormula struct {
+	// +kubebuilder:validation:MinLength=1
+	Formula string `json:"formula"`
+}
+
+// TimeseriesQuery is a discriminated union. Only Metrics are supported for autoscaling.
+// +kubebuilder:object:generate=true
+type DatadogPodAutoscalerTimeseriesQuery struct {
+	Metrics *DatadogPodAutoscalerMetricsTimeseriesQuery `json:"metrics,omitempty"`
+}
+
+// +kubebuilder:object:generate=true
+type DatadogPodAutoscalerMetricsTimeseriesQuery struct {
+	DataSource DatadogPodAutoscalerMetricsDataSource `json:"dataSource"`
+	// Optional variable name ("a", "b", etc.) to reference in formulas.
+	// +optional
+	Name *string `json:"name,omitempty"`
+	// Classic Datadog metrics query, e.g. "avg:system.cpu.user{*} by {env}".
+	// +kubebuilder:validation:MinLength=1
+	Query string `json:"query"`
+}
+
+// +kubebuilder:validation:Enum=metrics;cloud_cost
+type DatadogPodAutoscalerMetricsDataSource string
+
+const (
+	MetricsDataSourceMetrics   DatadogPodAutoscalerMetricsDataSource = "metrics"
+	MetricsDataSourceCloudCost DatadogPodAutoscalerMetricsDataSource = "cloud_cost"
```

Comment:
question: why is there `cloud_cost`? AFAIK we didn't consider supporting `cloud_cost` and wanted to support APM metrics instead

Reply:
This question is still pending

Reply:
Indeed, still need a change